### PR TITLE
[FrameworkBundle][Routing] Add link to source to router:match

### PR DESCRIPTION
**Pull request by l-vo**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | https://github.com/orgs/symfony/projects/1#card-30505860
| License       | MIT
| Doc PR        | 

Add link to source on `debug:router`/`router:match` when `framework.ide` is configured

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ CHANGELOG
 5.1.0
 -----
 
+ * Added link to source on controller on router:match/debug:router (when framework.ide is configured)
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md]: please add a few backticks

**Review by nicolas-grekas — APPROVED** [review]
nice thanks

**Review by chalasr — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php [hunk 2]
```diff
@@ -82,6 +82,11 @@ protected function describeRouteCollection(RouteCollection $routes, array $optio
 
     protected function describeRoute(Route $route, array $options = [])
     {
+        $defaults = $route->getDefaults();
+        if (!empty($defaults['_controller'])) {
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: We avoid using `empty` in core.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php (hunk 2 above)
Comment (l-vo) [reply src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: Fixed :)

**Comment by l-vo** [comment]
Added `$this->formatValue` to safely handle `$defaults['_controller']` if not a string

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php [hunk 3]
```diff
@@ -82,6 +82,14 @@ protected function describeRouteCollection(RouteCollection $routes, array $optio
 
     protected function describeRoute(Route $route, array $options = [])
     {
+        $defaults = $route->getDefaults();
+        if (isset($defaults['_controller'])) {
+            $defaults['_controller'] = $this->formatControllerLink(
+                $defaults['_controller'],
+                $this->formatValue($defaults['_controller'])
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: should be formatCallable actually

Comment (l-vo) [reply src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: I used formatValue to keep the current format but ok, formatCallback is the right way; updated

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php [hunk 4]
```diff
@@ -82,6 +82,14 @@ protected function describeRouteCollection(RouteCollection $routes, array $optio
 
     protected function describeRoute(Route $route, array $options = [])
     {
+        $defaults = $route->getDefaults();
+        if (isset($defaults['_controller'])) {
+            $defaults['_controller'] = $this->formatControllerLink(
+                $defaults['_controller'],
+                $this->formatCallable($defaults['_controller'])
+            );
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: Could be on one line.

Comment (l-vo) [reply src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: Fixed

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @l-vo.

**Pull request merged**